src/main/java/com/puppycrawl/tools/checkstyle/MultiThreadModeSettings.java

```diff
@@ -0,0 +1,89 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * A settings for the multi thread mode.
+ * @author Andrew Kuchev
+ */
+public class MultiThreadModeSettings implements Serializable {
```

Comment:
I dislike this class and variable names because it makes the reader assume multi-threading is always involved. This is not the case since we have methods like `getSingleThreadMode` where the instance can be a single thread.
How about we rename it to something like `ThreadModeSettings`?